api/apps/api/src/modules/legacy-project-import/domain/legacy-project-import-checker/legacy-project-import-checker.service.spec.ts

```diff
@@ -0,0 +1,123 @@
+import { FixtureType } from '@marxan/utils/tests/fixture-type';
+import { Test } from '@nestjs/testing';
+import { Either } from 'fp-ts/lib/Either';
+import { v4 } from 'uuid';
+import { LegacyProjectImportMemoryRepository } from '../../infra/legacy-project-import-memory.repository';
+import { LegacyProjectImport } from '../legacy-project-import/legacy-project-import';
+import { LegacyProjectImportStatuses } from '../legacy-project-import/legacy-project-import-status';
+import { LegacyProjectImportRepository } from '../legacy-project-import/legacy-project-import.repository';
+import {
+  LegacyProjectImportChecker,
+  legacyProjectImportDoesntExist,
+  LegacyProjectImportDoesntExist,
+} from './legacy-project-import-checker.service';
+import { MarxanLegacyProjectImportChecker } from './marxan-legacy-project-import-checker.service';
+
+let fixtures: FixtureType<typeof getFixtures>;
+
+beforeEach(async () => {
+  fixtures = await getFixtures();
+});
+
+it(`hasImportedLegacyProjectImport() should return legacyProjectImportdoesntExist if the project is not a legacy project`, async () => {
+  const res = fixtures.GivenProjectIsNotALegacyProject();
+
+  const result = await fixtures.WhenHasImportedLegacyProjectMethodIsCalled(res);
+
+  fixtures.ThenLegacyProjectImportDoesntExistIsReturned(result);
+});
+
+it.each(
+  Object.values(LegacyProjectImportStatuses).filter(
+    (kind) => kind !== LegacyProjectImportStatuses.Completed,
+  ),
+)(
+  `hasImportedLegacyProjectImport() should return false if given legacy project import has status %s`,
+  async (kind) => {
+    const id = await fixtures.GivenLegacyProjectImport(
+      LegacyProjectImportStatuses.AcceptingFiles,
```

Comment:
shouldn't this be
```suggestion
      kind,
```
?

Otherwise, we're not making use of the `kind` param we get by iterating over the statuses except the `Completed` one